docs/user/installation.rst

```diff
@@ -4,50 +4,239 @@
+<http://deeplearning.net/software/theano/install.html>`_ which cover installing
+the prerequisites for a range of operating systems, or ask for help on `our
+mailing list <https://groups.google.com/d/forum/lasagne-users>`_.
 
-Install from PyPI
-=================
 
-.. note:: Lasagne hasn't been released as of now.  The only way to
-          install it for now is from source.
+Prerequisites
+=============
 
-To install release 0.1 of Lasagne from PyPI, run the following
-commands:
+Python
+------
+
+Lasagne currently requires Python 2.7 or 3.4 to run. Please install Python via
+the package manager of your operating system if they are not included already.
+Use Python 2.7 if you're in doubt; it is still more widespread than Python 3.
+
+C compiler
+----------
+
+Theano requires a working C compiler, and numpy/scipy require a compiler as
+well if you install them via ``pip``. On Linux, the default compiler is usually
+``gcc``, and on Mac OS, it's ``clang``. Again, please install them via the
+package manager of your operating system.
+
+numpy/scipy + BLAS
+------------------
+
+Lasagne requires numpy of version 1.6.2 or above, and Theano also requires
+scipy 0.11 or above. Numpy/scipy rely on a BLAS library to provide fast linear
+algebra routines. They will work fine without one, but a lot slower, so it is
+worth getting this right (but less importantly so if you plan to use a GPU).
+
+If you install numpy and scipy via your operating system's package manager,
+they should link to the BLAS library installed in your system. If you install
+numpy and scipy via ``pip install numpy`` and ``pip install scipy``, make sure
+to have development headers for your BLAS library installed (e.g., the
+``libopenblas-dev`` package on Debian/Ubuntu) while running the installation
+command. Please refer to the `numpy/scipy build instructions
+<http://www.scipy.org/scipylib/building/index.html>`_ if in doubt.
+
+Theano
+------
+
+The version to install depends on the Lasagne version you choose, so this will
+be handled below.
+
+
+Stable Lasagne release
+======================
+
+.. note:: Lasagne hasn't been released yet. Please see the next section for
+    how to install the bleeding-edge version instead.
+
+Lasagne 0.1 requires a more recent version of Theano than the one available
+on PyPI. To install a version that is known to work, run the following command:
 
 .. code-block:: bash
 
   pip install -r https://raw.githubusercontent.com/Lasagne/Lasagne/0.1/requirements.txt
+
+A more recent version will often work as well, but at the time of writing, a
+simple ``pip install Theano`` will give you a version that is too old.
+
+To install release 0.1 of Lasagne from PyPI, run the following command:
+
+.. code-block:: bash
+
   pip install Lasagne==0.1
 
-Install from source
-===================
+If you do not have administrative rights on your machine, add ``--user`` to
+both commands to install into your home directory instead. To upgrade from an
```

Comment:
There was a very strong recommendation of using virtualenv or conda environments before. I removed it because I think it would make newbies uncomfortable to be told to better use something they've never heard of without explaining what it is, how to do it and why, and experienced users would think of using virtualenv on their own.
I'd welcome a more cautious hint in the beginning along the lines of "If you're developing different projects that may require different versions of numpy, scipy, or Theano, you might want to look into using virtualenv(link to a good guide on that) or conda environments(link to a good guide on that) instead of installing packages site-wide." I'm not using virtualenv myself, so I don't know what's a good formulation and what's a good introductory guide. Do you have any suggestions? Or do you want to send a PR after this one is merged?